backend/app/services/rag/retriever.py

```
@@ -0,0 +1,164 @@
+        k: int,
+        weight: float = 60.0
+    ) -> List[Document]:
+        """
+        Combine results using Reciprocal Rank Fusion (RRF)
+
+        RRF formula: score = sum(1 / (rank + k)) for each result list
+        """
+        doc_scores: Dict[str, Dict[str, Any]] = {}
+
+        # Process vector search results
+        for rank, (doc, score) in enumerate(vector_results):
```

Ruff (0.14.10)

Loop control variable `score` not used within loop body

Rename unused `score` to `_score`

(B007)